TypeSystem/variant.h

```diff
@@ -239,7 +265,7 @@ struct VariantImpl<NAME, TypeListImpl<TYPES...>> : CurrentVariantNameHelper<NAME
   }
 
   template <typename X>
-  std::enable_if_t<!std::is_same<X, CurrentStruct>::value, X&> VariantValueImpl() {
+  std::enable_if_t<!std::is_same<X, current::variant::object_base_t>::value, X&> VariantValueImpl() {
```

Comment:
Not sure I understand the question.

If it's about `enable_if` being part of the return type, not template declaration, I assume as there's no overloading involved, the current implementation would do.

If it's about the very presence of `enable_if`, it's to prohibit using the `object_base_t` as the type for `Exists<>` / `Value<>`.